### PR TITLE
parallelize mass-diff localization; report progress during mass-diff …

### PR DESCRIPTION
**Pull request by rmillikin**
…localization engine; add +- 3 missed mono search type for easier/better top-down search

### CODE REVIEW

File: EngineLayer/Localization/LocalizationEngine.cs [hunk 1]
```diff
@@ -26,36 +28,60 @@ public LocalizationEngine(IEnumerable<PeptideSpectralMatch> allResultingIdentifi
...
+                (range, loopState) =>
                 {
-                    // create new PeptideWithSetMods with unidentified mass difference at the given residue
-                    PeptideWithSetModifications peptideWithLocalizedMassDiff = peptide.Localize(r, massDifference);
+                    for (int i = range.Item1; i < range.Item2; i++)
+                    {
+                        PeptideSpectralMatch psm = unambiguousPsms[i];
+
+                        // Stop loop if canceled
+                        if (GlobalVariables.StopLoops) { break; }
```
Comment (rmillikin) [inline EngineLayer/Localization/LocalizationEngine.cs]: this code is the same as before except parallelized, with search progress reported

File: GUI/SearchTaskWindow.xaml [hunk 2]
```diff
@@ -815,6 +815,13 @@
                                                     </TextBlock>
                                                 </RadioButton.ToolTip>
                                             </RadioButton>
+                                            <RadioButton x:Name="massDiffAcceptPlusOrMinusThree" Content="+- 3 Missed Monoisotopic Peaks" ToolTipService.ShowDuration="999999" ToolTipService.InitialShowDelay="500">
```
Comment (rmillikin) [inline GUI/SearchTaskWindow.xaml]: this adds a new mass diff acceptor. plus or minus three daltons. it's good for top-down searches where the missed monos are in both directions

**Comment by acesnik** [comment]
Looks good. I wonder how much more search time the +/- 3 Da setting adds for TD searches.

**Comment by rmillikin** [comment]
not much extra time, I think. most of the time is spent reading/"digesting" the database (i.e., creating the proteoform search space) and localizing mass-differences, if an open-mass search is performed. fragmenting and searching for a theoretical proteoform is not too expensive

**Pull request merged**